src/tester.rs

```diff
@@ -146,6 +157,31 @@ impl LangTester {
         self
     }
 
+    /// Specify a function which sets options for the [`fm`](https://crates.io/crates/fm) library.
+    /// `fm` is used for the fuzzy matching in `lang_tester`. This function can be used to override
+    /// `fm`'s defaults for a given file, a given testing stream (stderr or stdout): it is passed a
+    /// [`FMBuilder`](https://docs.rs/fm/*/fm/struct.FMBuilder.html) and must return a `FMBuilder`.
+    /// For example, to make use of `fm`'s "name matcher" option such that all instances of `$1`
+    /// must match the same value (without precisely specifying what that value is) one could use
+    /// the following:
+    ///
+    /// ```rust,ignore
+    /// LangTester::new()
+    ///    ...
+    ///    .fm_options(|_, _, fmb| {
+    ///        let ptn_re = Regex::new(r"\$.+?\b").unwrap();
+    ///        let text_re = Regex::new(r".+?\b").unwrap();
+    ///        fmb.name_matcher(Some((ptn_re, text_re)))
+    ///    })
+    /// ```
+    pub fn fm_options<F>(&mut self, fm_options: F) -> &mut Self
+    where
+        F: 'static + for<'a> Fn(&'a Path, TestStream, FMBuilder<'a>) -> FMBuilder<'a> + Send + Sync,
```

Comment:
Do we need to document the first two parameters of the closure? There's a path and a `TestStream` that I'm unsure what are for.

Reply:
You're right, they should be documented, but if you can't guess what they are with 100% accuracy (once you've looked up what `TestStream` is), I'd be surprised ;)

Reply:
Actually they are documented!

```
This function can be used to override `fm`'s defaults for a given file, a given testing stream (stderr or stdout)
```

Reply:
> This function can be used to override `fm`'s defaults for a given file, a given testing stream

So that's a file *and* a testing stream (since they are not `Option`). So what path would you pass for e.g. stdout? Ideally it should be clear from this doc-string without having to look elsewhere.

Reply:
Does https://github.com/softdevteam/lang_tester/pull/63/commits/f43fb87b841ede24dce246152f1ce28e6566faaa help?

Reply:
Ah, so the `Path` is to an executable which is run to get stderr/stdout?

I see.

Reply:
No, it's the test file (exactly the same as for `test_cmds` and so on).

Reply:
I think I get it, but was confused by:
> when executing that file

I guess someone/something is compiling the test file. As far as I can see they are just Rust programs with special comments, which can't be directly executed. Maybe I shouldn't take it so literally.

Reply:
Try https://github.com/softdevteam/lang_tester/pull/63/commits/0e37c052eb45cfd434fed34c2da135786a2fc931.

Reply:
OK.

Sorry if i was pedantic. I've never used lang_tester, so I'm probably out of the loop.

Reply:
Not a problem!